packages/flutter_tools/test/integration.shard/analyze_once_test.dart

```diff
@@ -339,7 +339,7 @@ int analyze() {}
     );
   });
 
-  testWithoutContext('analyze once only fatal-infos has warning issue finally exit code 1.', () async {
+  testWithoutContext('analyze once only fatal-infos has warning issue finally exit code 0.', () async {
```

Comment:
This looks like a good change. Can you add a test that verifies that there will be a 1 exit code with `--fatal-warnings` provided?